packages/xo-web/src/common/render-xo-item.js

```diff
@@ -60,147 +108,206 @@ export const VmItem = decorate([
-            <span className='text-muted'> - {container.name_label}</span>
-          )}
-          {isSrWritable(sr) && (
-            <span>{` (${formatSize(sr.size - sr.physical_usage)} free)`}</span>
-          )}
-        </span>
-      )}
-    </XoItem>
-  ),
+  ({ template }) => {
+    if (template === undefined) {
+      return UNKNOWN_ITEM
+    }
+
+    return (
+      <span>
+        <Icon icon='vm' /> {template.name_label}
+      </span>
+    )
+  },
 ])
 
-SrItem.propTypes = XO_ITEM_PROP_TYPES
+VmTemplate.propTypes = {
+  id: PropTypes.string.isRequired,
+}
 
-export const RemoteItem = decorate([
-  addSubscriptions(({ id }) => ({
-    remote: cb =>
-      subscribeRemotes(remotes => {
-        cb(get(() => remotes.find(remote => remote.id === id)))
-      }),
-  })),
-  ({ remote, ...props }) => (
-    <XoItem item={remote} to='/settings/remotes' {...props}>
-      {() => (
-        <span>
-          <Icon icon='remote' /> {remote.name}
-        </span>
-      )}
-    </XoItem>
-  ),
+VmTemplate.defaultProps = {
+  link: false,
+  newTab: false,
+}
+
+// ===================================================================
+
+export const Sr = decorate([
+  connectStore(() => {
+    const getSr = createGetObject()
+    const getContainer = createGetObject(
+      createSelector(getSr, sr => get(() => sr.$container))
+    )
+    return (state, props) => ({
+      // FIXME: props.self ugly workaround to get object as a self user
+      sr: getSr(state, props, props.self),
+      container: getContainer(state, props),
+    })
+  }),
+  ({ sr, container, link, newTab }) => {
+    if (sr === undefined) {
+      return UNKNOWN_ITEM
+    }
+
+    return (
+      <LinkWrapper link={link} newTab={newTab} to={`/srs/${sr.id}`}>
+        <Icon icon='sr' /> {sr.name_label}
+        {container !== undefined && (
+          <span className='text-muted'> - {container.name_label}</span>
+        )}
+        {isSrWritable(sr) && (
+          <span>{` (${formatSize(sr.size - sr.physical_usage)} free)`}</span>
+        )}
+      </LinkWrapper>
+    )
+  },
 ])
 
-RemoteItem.propTypes = XO_ITEM_PROP_TYPES
+Sr.propTypes = {
+  id: PropTypes.string.isRequired,
+  link: PropTypes.bool,
+  newTab: PropTypes.bool,
+}
 
-export const PoolItem = decorate([
-  connectStore(() => ({
-    pool: createGetObject(),
-  })),
-  ({ pool, ...props }) => (
-    <XoItem item={pool} to={`/pools/${get(() => pool.id)}`} {...props}>
-      {() => (
-        <span>
-          <Icon icon='pool' /> {pool.name_label || pool.id}
-        </span>
-      )}
-    </XoItem>
-  ),
+Sr.defaultProps = {
+  link: false,
+  newTab: false,
+}
+
+// ===================================================================
+
+export const Vdi = decorate([
+  connectStore(() => {
+    const getObject = createGetObject()
+    // FIXME: props.self ugly workaround to get object as a self user
+    return (state, props) => ({
+      vdi: getObject(state, props, props.self),
```

Comment:
You can make it inline.

Reply:
You mean inline `getObject`? We want to use the same selector for all the component's lifetime.

Reply:
My bad, you're right, we don't want create a function on each call.